testing/test_invocations.py

```diff
@@ -287,3 +290,39 @@ def hello(self, arg):
     pm.register(Plugin())
 
     assert pm.hook.hello(arg=10, extra=20) == ["Plugin.hello"]
+
+
+def test_non_wrapper_generator(pm: PluginManager) -> None:
+    """A hookimpl can be a generator without being a wrapper,
+    meaning it returns an iterator result."""
+
+    class Api:
+        @hookspec
+        def hello(self) -> Iterator[int]:
+            raise NotImplementedError()
```

Comment:
We could add `raise NotImplementedError` to `.coveragerc` as a line to be excluded from coverage.